src/lib/fetch.ts

```diff
@@ -1086,7 +1086,17 @@ async function checkResponseStatus(
         undefined,
         {
           statusCode: response.status,
-          statusText: jsonResponse.errors,
+          statusText: jsonResponse.errors
+            .map((error: any) => error.message)
+            .join('\n'),
+        } as FetchError,
+      ];
+    } else if (jsonResponse.data === undefined) {
+      return [
+        undefined,
+        {
+          statusCode: response.status,
+          statusText: `GraphQL response data is undefined`,
```

Comment:
just double-checking -- are we sure that `undefined` response data always means that the request failed?

Reply:
Ok I guess we can look at our current queries to check that we always want something returned, and this is currently the case 👍🏻